core/server/worker/src/main/java/alluxio/worker/block/AsyncCacheRequestManager.java

```diff
@@ -107,6 +108,17 @@ public void submitRequest(Protocol.AsyncCacheRequest request) {
    */
   private boolean cacheBlockFromUfs(long blockId, long blockSize,
       Protocol.OpenUfsBlockOptions openUfsBlockOptions) {
+    // Check if the block has already been cached on this worker
+    long lockId = mBlockWorker.lockBlockNoException(Sessions.ASYNC_CACHE_SESSION_ID, blockId);
+    if (lockId != BlockLockManager.INVALID_LOCK_ID) {
+      try {
+        mBlockWorker.unlockBlock(lockId);
+      } catch (BlockDoesNotExistException e) {
+        LOG.error("Unlock block failed on caching block from UFS. We should never reach here", e);
+      }
+      return true;
+    }
+    // Check if the block has been requested in UFS block store
```

Comment:
Should we do this outside of cacheBlockFromUfs, since this also helps filter blocks which need to be read from another worker?

Reply:
done. put it in `submitRequest`